manager/scheduler/scheduler.go

```diff
@@ -243,10 +243,16 @@ func (s *Scheduler) deleteTask(ctx context.Context, t *api.Task) {
 
 func (s *Scheduler) createOrUpdateNode(n *api.Node) {
 	var resources api.Resources
+	nodeInfo := s.nodeHeap.nodeInfo(n.ID)
 	if n.Description != nil && n.Description.Resources != nil {
 		resources = *n.Description.Resources
+		// reconcile resources
+		for _, task := range nodeInfo.Tasks {
+			reservations := taskReservations(task.Spec)
+			resources.MemoryBytes -= reservations.MemoryBytes
```

Comment:
I really think we need to do our accounting outside of the Description field. Writing to `Description` is part of session registration. Having multiple writers for the same dataset is always a disaster.

It seems we'll need a secondary place on the node to keep track of reservations.